opensquirrel/utils/matrix_expander.py

```diff
@@ -95,19 +95,22 @@ def __init__(self, qubit_register_size: int) -> None:
         self.qubit_register_size = qubit_register_size
 
     def visit_bloch_sphere_rotation(self, rot: BlochSphereRotation) -> NDArray[np.complex_]:
-        assert rot.qubit.index < self.qubit_register_size
+        if rot.qubit.index >= self.qubit_register_size:
+            raise IndexError("Index out of range.")
```

Comment:
I'm more than fine with these changes. I know Pablo differentiated between using assert and exceptions. In C++ code, asserts are a debug mechanism. They are kind of extra checks. If you have a runtime error, and you debug it, these extra checks give you more information. But I am not really used to program with asserts, and I prefer these let's say hard checks that are done also in release mode, via exceptions.